### ISSUE TITLE
Allow multi values on filters (rest, cli?)

### ISSUE THREAD

**Issue by dbavery** [StackStorm/st2#2953]
as a restconf user, I would like to be able to specify multiple values for filter criteria when I make a rest call, so I can streamline my requests and reduce overhead of multiple requests.

For example,

The specific use-case I have is that I have a list of executions, which I started, and I need to check the execution ids to see if they have been completed. Right now I have to request the status of each one individual, or request way more than I want (say previous 100 and hope that they are in that list).

Instead, I would like to be able to send the list into the request some how.

For example

/api/v1/executions/?id=<id1>,<id2> ...

or perhaps allow the request to be specified in a json body:

{
    "idlist":[ "id1", "id2" ...]
}

While my specific use-case is for executions, I can see a general benefit to allow all filters to allow multiple values for filters. It moves the overhead of performing the filter closer to the source data which ultimately means more efficient downstream (less data over network etc).

**Comment by Kami** [comment]
I do think we should allow user to provide multiple values for some filters like an ID and in fact, we already support that in some places.

HTTP already solves "specifying multiple values for a parameter" part by allowing user to specify the same query parameter multiple times (e.g. `?id=1&id=2&id=3`), but sadly in some places we already use comma delimited string for the parameter value (e.g. `?foo=a,b,c`) so for consistency sake we should probably go with a comma delimited value for now (and eventually we can / should fix it to be consistent with HTTP spec).

In short - good idea, you are welcome to take a stab at the code yourself (the change should be relatively straight forward).

**Comment by Kami** [comment]
@dbavery I've merged #2958 which should resolve that - please have a look and confirm.

If confirmed, I will go ahead and close this issue.

**Comment by dbavery** [comment]
@Kami Assuming I am reading the code right (new to stack storm) it appears that this was generically added across the board to all rest calls? If that assumption is correct then I agree that this solves the direct ask. Ideally we would add this capability to all filter options everywhere, but that is potentially an open-ended work item with less that clear use-cases.

If my assumption that filtering by multiple IDs has been applied to all tables is correct, then I would agree we can close this out.

Thank you!

**Comment by enykeev** [comment]
Looks that way to me. Closing.
